SourceCode/Industrial_Heat/ftt_mtm_main.py

```diff
@@ -268,12 +267,13 @@ def solve(data, time_lag, iter_lag, titles, histend, year, domain):#, #specs, co
 
             data_dt[var] = copy.deepcopy(time_lag[var])
 
-        # Create the regulation variable #Regulate capacity #no regulations yet, isReg full of zeros
+        # Create the regulation variable #Regulate ued #no regulations yet, isReg full of zeros
         isReg = np.zeros([len(titles['RTI']), len(titles['ITTI'])])
-        division = np.zeros([len(titles['RTI']), len(titles['ITTI'])])
-        division = divide((data_dt['IWK3'][:, :, 0] - data['IRG3'][:, :, 0]),
-                          data_dt['IRG3'][:, :, 0])
-        isReg = 0.5 + 0.5*np.tanh(2*1.25*division)
+        isReg = np.where(data['IRG3'][:, :, 0] > 0.0,
+                          (np.tanh(1 +
+                              (data_dt['IUD3'][:, :, 0] - data['IRG3'][:, :, 0]) 
+                                  / data['IRG3'][:, :, 0])),
+                          0.0)
         isReg[data['IRG3'][:, :, 0] == 0.0] = 1.0
```

Comment:
isReg is now a value between -1 and 1 everywhere. is that intentional? I believe it should be between 0 and 1, in a shape like 0.5 + 0.5 tanh(a + bx). If you put a=1.5 and b=10, you get quite reasonable behaviour.

Reply:
isReg seems to be different in all versions, now I look at it. I did correct isReg to have the right variables in, but didn't look at the definitions. 

However, tanh( 1 + (C-R)/R) cannot become negative if both capacity C and regulation R are positive. It will go between 0 and 1. This is because the fraction part of this calculation cannot be greater than 1. 

Let's say capacity is 10 and the regulation is set to 20. Tanh(1 + (10-20)/20) = tanh(0.5) = 0.46
Let's say capacity is 10 and the regulation is set to 5. Tanh(1 + (10-5)/5) = tanh(2) = 0.96
For very, very small R, isreg will get closer and closer to 1.
For very, very large R relatve to C, isreg will be close to zero.

Reply:
Apologies for not catching the differences!

Reply:
I knew I was overlooking something (the [1+ ..] that ensures tanh doesn't drop below 1 ). 

I still think this is insufficiently strict when CAP>=REG (isReg = 0.76). We expect almost all new investments to be stopped at that point. The equation I fiddled out stops 95% of the investments at that point (0.5 + 0.5*tanh(1.5 + 10*[C-R]/R) ). The comments in the old code said 90% should be stopped. My equation is a bit ugly however, with quite random numbers.

Reply:
I'm happy to go with that equation if you have tested it, there is nothing wrong with ugly maths if it is ensuring the model works as intended.

I can't change my commit, but it looks like this will work for all the models (just checked it made sense for heat, where HREG is in shares for some reason) so if you want to put it in as a suggested change in all of them, I'm happy to add it to this pull!

Reply:
Changes made.